WIP: first go at newline support

esell opened the pull request:
This is likely pretty horrible but I thought I'd give it a go. A few things are broken and I haven't quite figured them out:

- With newline enabled, there is hanging "block" on part of the prompt
- Right now you need to pass `--newline 1/0`. really it should just need the `--newline` flag

jD91mZM2 commented:
It looks like you unfortunately reformatted the whole thing. I would prefer sticking with the current layout, or at least doing it in a separate pull request. I know that it's going to be a pain to undo that though, and I'm afraid I can't help you there. That's one of the reasons I stopped auto formatting.

For the last thing, you can probably just remove `takes_value` and check if the match exists at all.

esell commented:
Fixed the formatting, sorry about that. I changed the `newline` parameter also. Still looking at why the extra block is added...

If you don't like the way things are going feel free to say "no" haha.

jD91mZM2 commented on src/main.rs:
```diff
@@ -210,7 +210,13 @@ fn main() {
+           Module::Newline => {
+                let (mut bg, mut fg) = (theme.username_bg, theme.username_fg);
+                if matches.is_present("newline")  {
+                    segments.push(Segment::new(bg, fg, String::new()).set_newline());
```
You're overcomplicating it. You only need  
```
if matches.is_present("newline") {
    println!();
}
```
at the end of the entire function.  
No need to add a new module.

esell replied:
Just an FYI, with this guy I couldn't get the newline to pop-up. I swapped it to `println!("\\n");` though and it worked (I'm using bash). I appreciate all the help with this, it was a good learning experience even though you ended up writing the final solution haha.

jD91mZM2 replied:
I'm sorry if I took over your fun :(

esell replied:
Oh, no way! Sorry if it came across that way, it was a great way to learn and seeing the end solution helped me understand a lot more than when I started.

jD91mZM2 replied:
Ah, okay! Awesome!

jD91mZM2 commented on src/main.rs:
```diff
@@ -235,3 +241,4 @@ fn main() {
         flame::dump_html(&mut File::create("profile.html").unwrap()).unwrap();
     }
 }
+
```
Extra newline here isn't needed.

jD91mZM2 commented:
Awesome, now the code is as minimal as it has to be. 😄  
If this is what you wanted, tell me to merge.  
Otherwise, tell me and I'll undo my changes.
